contributing-docs/17_adding_api_endpoints.rst

```diff
@@ -18,122 +18,78 @@
+3. Persisted openapi spec is located in ``v1-generated.yaml`` and a hook will take care of updating it based on your new Endpoint, you just need to add and commit the change.
 4. Run the following command to execute all pre-commit hooks:
 
 .. code-block:: bash
 
    pre-commit run --all-files
 
 
-Optional: Adding Schemas
------------------------------
-In some cases, you may need to define additional schemas for new data structures. For example, if you are adding an endpoint that involves new data objects or collections, you may define a schema in a Python file. Here's an example:
+Optional: Adding Pydantic Model
+-------------------------------
+In some cases, you may need to define additional models for new data structures. For example, if you are adding an endpoint that involves new data objects or collections, you may define a model in a Python file. The model will be used to validate and serialize/deserialize objects. Here's an example:
 
 .. code-block:: python
 
-   class TaskLogPageSchema(Schema):
-       """Schema for task log pagination details."""
-
-       total_pages = fields.Int(description="Total number of pages for task logs.")
-       current_page = fields.Int(description="Current page number.")
-       page_size = fields.Int(description="Number of logs per page.")
-
-These schemas are defined to structure and validate the data handled by the API. Once defined, you can include these schemas in the OpenAPI specification file (e.g., v1.yaml) and reference them in the API endpoint definitions.
-
-For example, in v1.yaml, you might add:
-
-.. code-block:: yaml
+    class DAGModelResponse(BaseModel):
+        """DAG serializer for responses."""
 
-   components:
-     schemas:
-       TaskLogPage:
-         type: object
-         properties:
-           total_pages:
-             type: integer
-             description: Total number of pages for task logs.
-           current_page:
-             type: integer
-             description: Current page number.
-           page_size:
-             type: integer
-             description: Number of logs per page.
+        dag_id: str
+        dag_display_name: str
+        is_paused: bool
+        is_active: bool
+        last_parsed_time: datetime | None
 
-Including schemas helps in automatically generating API documentation and ensures consistent data structures across the API.
+These models are defined to structure and validate the data handled by the API. Once defined, these models will automatically be added to the OpenAPI spec file as long as they are actually used by one endpoint.
 
-After adding or modifying schemas, make sure to run the pre-commit hooks again to update any generated files.
+After adding or modifying Pydantic models, make sure to run the pre-commit hooks again to update any generated files.
 
-Note on the New FastAPI API
----------------------------
+Situational: Legacy Endpoint Migration
```

Comment:
If we migrate the public API, should we also make sure that existing pytests are working 1:1 with the new endpoint?

Reply:
Good point I have added some notes about that.

Yes the strategy right now is basically to copy (and adapt) the associated tests of the endpoint legacy endpoint to the new one. 

We cannot purely 'reuse' what's in place for flask because the test client is different between flask and fastapi, and we would have to 'abstract' the test clients and fixtures to be able to use the current flask tests and plug them on fastapi. Which would be tedious, I don't know if this is even possible so at the moment duplicating and adapting for fastapi is easier and flask will get removed soon after.